### PR TITLE
add testscript CLI tests; remove examples

### PR DESCRIPTION
**Pull request by jmank88**
This PR introduces https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript CLI tests as `testdata/scripts/*.txtar` files. These supersede the `core/main_test.go` examples, and offer better coverage both in terms of variety of cases (easy to do more than just `--help` output) as well as completeness (invokes a real binary instead of a test hook). These tests still execute as part of the normal `go test` suite, but they can also be interacted with via a [CLI tool](https://pkg.go.dev/github.com/rogpeppe/go-internal/cmd/testscript#section-readme), to run them (with the binary of your choice) but also to do a few other useful things, like _automatically regenerate/update the expected output of the tests_, relieving the maintenance burden of large refactors.

##### Blog Series

1. https://bitfieldconsulting.com/golang/test-scripts
2. https://bitfieldconsulting.com/golang/cli-testing
3. https://bitfieldconsulting.com/golang/test-scripts-files
4. https://bitfieldconsulting.com/golang/conditions-concurrency
5. https://bitfieldconsulting.com/golang/testscript-tool

### CODE REVIEW

File: core/services/chainlink/config_general.go [hunk 1]
```diff
@@ -232,9 +232,12 @@ func validateEnv() (err error) {
...
 func (g *generalConfig) LogConfiguration(log coreconfig.LogFn) {
-	log("Secrets:\n", g.secretsTOML)
-	log("Input Configuration:\n", g.inputTOML)
-	log("Effective Configuration, with defaults applied:\n", g.effectiveTOML)
+	log("Secrets:")
+	log(g.secretsTOML)
+	log("Input Configuration:")
+	log(g.inputTOML)
+	log("Effective Configuration, with defaults applied:")
+	log(g.effectiveTOML)
```
Comment (jmank88) [inline core/services/chainlink/config_general.go:240]: The old way was introducing some extra whitespace.

File: GNUmakefile [hunk 2]
```diff
@@ -85,6 +85,16 @@ go-solidity-wrappers-ocr2vrf: pnpmdep abigen ## Recompiles solidity contracts an
 generate: abigen codecgen mockery ## Execute all go:generate commands.
 	go generate -x ./...
 
+.PHONY: testscripts
+testscripts: ## Install and run testscript against testdata/scripts/* files.
+	go install github.com/rogpeppe/go-internal/cmd/testscript@latest
+	go install -ldflags "-X github.com/smartcontractkit/chainlink/v2/core/static.Version=unset -X github.com/smartcontractkit/chainlink/v2/core/static.Sha=unset"
```
Comment (jmank88) [inline GNUmakefile:91]: This works, but changing the local installation permanently is not ideal.
I tried to just `go build` in the working dir and target via `PATH` or `alias`, but then I get:
> [exec: "chainlink": cannot run executable found relative to current directory]

Comment (jmank88) [reply GNUmakefile]: Fixed with `$(CURDIR)` instead of `.`.

**Comment by jmank88** [comment]
CI output looks successful, but is producing `exit code 2` :confused:

**Comment by jmank88** [comment]
> CI output looks successful, but is producing `exit code 2` confused

Reproduced locally with `-coverprofile`. Fixed in latest `1.10.0`.

**Review by krehermann — COMMENTED** [review]
how can we leverage this to catch or mitigate circumstances like

https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1681771962001029?thread_ts=1681334193.100579&cid=C0117GGJB6Y

is there a test in this PR that would cover that?

as a followup, it would be very interesting to measure our coverage of cli cmds and permutations in these test scripts.

**Comment by jmank88** [comment]
> how can we leverage this to catch or mitigate circumstances like
> 
> [chainlink-core.slack.com/archives/C0117GGJB6Y/p1681771962001029?thread_ts=1681334193.100579&cid=C0117GGJB6Y](https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1681771962001029?thread_ts=1681334193.100579&cid=C0117GGJB6Y)
> 
> is there a test in this PR that would cover that?
> 
> as a followup, it would be very interesting to measure our coverage of cli cmds and permutations in these test scripts.

This PR doesn't cover that directly but a test can be added pretty easily.

**Comment by kalverra** [comment]
Can this be added to CI tests? Or is that already covered?

**Comment by jmank88** [comment]
> Can this be added to CI tests? Or is that already covered?

The `make testscript` command is actually secondary. They are wired in to run as regular go tests here in main_test.go: https://github.com/smartcontractkit/chainlink/pull/8989/files#diff-79ce3229c13921b79b1175dcb211336aaf84dfd8edcf19c07698934d4fe70e5e

**Review by kalverra — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

**Pull request merged**